manifests/agent.pp

```diff
@@ -432,24 +432,16 @@
   }
 
   # Controlling the 'zabbix-agent' service
-  if str2bool(getvar('::systemd')) {
-    $service_provider = 'systemd'
-    $service_path = undef
-  } elsif $facts['os']['name'] == 'AIX' {
-    $service_provider = 'init'
-    $service_path = '/etc/rc.d/init.d'
-  } else {
-    $service_provider = undef
-    $service_path = undef
-  }
   service { $servicename:
-    ensure     => $service_ensure,
-    enable     => $service_enable,
-    path       => $service_path,
-    provider   => $service_provider,
-    hasstatus  => true,
-    hasrestart => true,
-    require    => Package[$zabbix_package_agent],
+    ensure  => $service_ensure,
+    enable  => $service_enable,
+    require => Package[$zabbix_package_agent],
+  }
+
+  # Override the service provider on AIX
+  # Doing it this way allows overriding it on other platforms
+  if $facts['os']['name'] == 'AIX' {
+    Service[$servicename] { service_provider => 'init' }
```

Comment:
given this documentation: https://puppet.com/docs/puppet/6.18/types/service.html#service-provider-init

```ruby
true == begin
      os = Facter.value(:operatingsystem).downcase
      family = Facter.value(:osfamily).downcase
      !(os == 'debian' || os == 'ubuntu' || family == 'redhat')
  end
```

i'm sceptical that puppet will know the correct `$service_path`, so maybe let's bring this back.

Reply:
I still dont get why we need to set the provider explicitly :(

Reply:
that's why I'm hoping @bjschafer can answer that.

Reply:
because the default `service` provider is [src](https://puppet.com/docs/puppet/6.18/types/service.html#service-provider-src), but the zabbix package, i assume, provides an init style service file

Reply:
I did see that, but I don't know if `src` can run init style service files.